File: dataset/human/c/file_1906.c
#include <stdio.h>

int main(){
    int N,ans=0;
    scanf("%d",&N);
    int A[N],B[N],C[N-1];
    for(int i=0;i<N;i++){
        scanf("%d",&A[i]);
    }
    for(int i=0;i<N;i++){
        scanf("%d",&B[i]);
    }
    for(int i=0;i<N-1;i++){
        scanf("%d",&C[i]);
    }
    for(int i=0;i<N;i++){
        ans+=B[i];
    }
    for(int i=0;i<N-1;i++){
        if(A[i+1]==A[i]+1){
            ans+=C[A[i]-1];
        }
    }
    printf("%d",ans);
    return 0;
}

File: dataset/human/python/file_1906.py
N, M = map(int, input().split())

A = [""] * N
for _ in range(N):
    A[_] = input()

B = [""] * M
for _ in range(M):
    B[_] = input()

import sys
ans = True

for nvi in range(N-M+1):
    for nhi in range(N-M+1):
        ok = True

        for vi in range(M):
            for hi in range(M):
                if A[nvi+vi][nhi+hi] == B[vi][hi]:
                    pass
                else:
                    ok = False
                    break
        
        if ok:
            print("Yes")
            sys.exit()

print("No")
